traits/tests/test_enum.py

```diff
@@ -296,3 +302,25 @@ def test_base_enum(self):
         with self.assertRaises(TraitError):
             obj.slow_enum = "perhaps"
         self.assertEqual(obj.slow_enum, "no")
+
+
+class TestGui(GuiTestAssistant, unittest.TestCase):
+
+    def test_create_editor(self):
+        obj = EnumCollectionGUIExample()
+        traits = obj.class_trait_names()
+        traits.remove("trait_added")
+        traits.remove("trait_modified")
+
+        ui = obj.edit_traits()
```

Comment:
I agree that the `inverse_mapping` is detail that traits can ignore. But `mapping` looks to me a reasonable public attribute exposed by traitsui for introspection.
Exercising GUI components and its synchronization with traits would be even further into the implementation details: It requires the GUI toolkit for a start and it would be more of an integration test than just introspecting the mapping the editor has received.

Reply:
To be fair, traits should not have depended on traitsui.  These `create_editor` methods bring in a dependency traits should not have. The default editors definitions should live in traitsui, not here. Taking that context into account, it is good enough to have one test that just exercises the instantiation of editors (achieved by the `edit_traits` here) for `Enum` with `values` or with `name` to bring in the test coverage. The rest is really not traits's business...

Reply:
Agreed: we want to keep the influence of TraitsUI on Traits as small as possible. Ideally, at some point in the future we'd set up a registry mechanism in TraitsUI for registering default editors for various traits, then we could remove this code entirely. Not high priority, though.